quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnameV2PropertyMappers.java

```diff
@@ -41,6 +46,50 @@ public static void validateConfig(Picocli picocli) {
         if (!inUse.isEmpty()) {
             picocli.warn("Hostname v1 options %s are still in use, please review your configuration".formatted(inUse));
         }
+        boolean isProd = Environment.PROD_PROFILE_VALUE.equals(org.keycloak.common.util.Environment.getProfile());
+        boolean httpsEnabled = HttpPropertyMappers.isHttpsEnabled();
+        String host = Configuration.getConfigValue(HostnameV2Options.HOSTNAME).getValue();
+        if (host != null && validateFullHostname(httpsEnabled, isProd, host, picocli)) {
+            return;
+        }
+        if (httpsEnabled) {
+            // must be tls passthrough, but there's no way of knowing the intention
+            return;
+        }
+        String proxyHeaders = Configuration.getConfigValue(ProxyOptions.PROXY_HEADERS).getValue();
+        if (proxyHeaders != null) {
+            // must not be tls passthrough, but there's no way of knowing the intention
+            return;
+        }
+        if (isProd) {
+            if (host == null) {
+                picocli.warn("With HTTPS not enabled and hostname-strict=false, cross-origin cookies will not be allowed.");
+            } else if (!SecureContextResolver.isLocal(host)) {
+                picocli.warn("HTTPS is not enabled on the server, either the `hostname` should be set to a full URL or `proxy-headers` should be used. This is likely a misconfiguration.");
+            } // else warn on prod
```

Comment:
Why `// else warn on prod`? The warning si already there, no?

Reply:
The else case here is something like hostname=http://localhost - doesn't seem like that would be used in production, but I wasn't sure if it warrented a warning.